homeassistant/components/cover/mqtt.py

```diff
@@ -469,6 +469,11 @@ def find_percentage_in_range(self, position, range_type=TILT_PAYLOAD):
         offset_position = position - min_range
         position_percentage = round(
             float(offset_position) / current_range * 100.0)
+
+        max_percent = 100
+        min_percent = 0
+        position_percentage = min(max(position_percentage, min_percent),
+                                  max_percent)
         if range_type == TILT_PAYLOAD and self._tilt_invert:
```

Comment:
Is there an existing test for this case?

Reply:
Lines 317 for regular and 378 for inverted in test_mqtt.py

Reply:
@MartinHjelmare maybe let's make another PR about tilts after this one.
I see many inconsistencies in tilts.
I wrote some of them down:
File MQTT.py
- `tilt_updated` expects value within range `<tilt_min-tilt_max>` and converts it to percentage. What about open/close position? Shouldn't `_tilt_open_position`, `_tilt_closed_position`  be withing range `<min-max>`? (I don't understand why there are 4 values - min,max,open,close - do you know why?)
- `async_open_cover_tilt` - in optimistic mode this sets value as _tilt_open_position, which doesn't have to be percentage value within <min-max> range - **incostistency** with `tilt_updates`
- `async_close_cover_tilt` - in optimistic mode this sets value as _tilt_closed_position, which doesn't have to be percentage value within <min-max> range- **incostistency** with `tilt_updates`
- `async_set_cover_tilt_position` - expects always percentage value - this is **OK**
- `async_set_cover_tilt_position` - **missing optimistic mode**
- `current_cover_tilt_position` in normal mode returns percentage value and in optimistic returns value of tilt_opened/tilt_closed - different behaviour in two modes.
- `find_percentage_in_range` and `find_in_range_from_percent`, I can't find any reason of this substraction if tilt_inverted.  Docs says: `higher values toward closed and lower values toward open.`

|                                                    | Min | Max | Invert | Position | Out value | Expected value |
|----------------------------------------------------|-----|-----|--------|----------|----------:|----------------|
| set_cover_tilt_position find_in_range_from_percent | 0   | 100 | false  | 40       |        40 | 40             |
| set_cover_tilt_position find_in_range_from_percent | 0   | 100 | true   | 40       |        60 | 60             |
| set_cover_tilt_position find_in_range_from_percent | 100 | 0   | false  | 40       |        60 | 60             |
| set_cover_tilt_position find_in_range_from_percent | 100 | 0   | true   | 40       | **_40_**  <- don't understand this value      | **60**             |

In test_mqtt.py I see some things which are derived from mqtt.py like:
- `test_tilt_via_topic`
    - 'tilt_opened_value': 400,
    - 'tilt_closed_value': 125
    - max is set as default 100, min as default 0,
    - `fire_mqtt_message(self.hass, 'tilt-status-topic', '50')` - why can it be out of range tilt_opened_value/tilt_closed_value? Is it working different way than mqtt cover, so set_cover_position is sending percentage value to mqtt tilt?

Reply:
@MartinHjelmare what do you think about above?

Reply:
I think we allow different values for `tilt_min` + `tilt_max` and `tilt_closed_value` + `tilt_opened_value` cause they are used for different topics. The first two are used for the tilt status topic. The latter two are used for the tilt command topic. A device doesn't necessarily support both.

We should always convert the internal home assistant tilt position to a percent value. :+1: 

I'm not sure about the inversion of tilt yet.

I agree that changes to tilt should be done in a separate PR.